apps/gp2-server/test/data-providers/project.data-provider.test.ts

```diff
@@ -112,83 +113,129 @@ describe('Project Data Provider', () => {
+          member!.user = undefined!;
+          Project.flatData.members = [member];
+          const { members } = parseProjectToDataObject(Project);
+          expect(members).toEqual([]);
+        });
+        test('pm emails are added if available', () => {
+          const email = 'tony@starkenterprises.com';
+          const project = getGraphQLProject();
+          project.flatData.pmEmail = email;
+          const { pmEmail } = parseProjectToDataObject(project);
+          expect(pmEmail).toEqual(email);
+        });
+
+        test('lead emails are added if available', () => {
+          const email = 'tony@starkenterprises.com';
+          const project = getGraphQLProject();
+          project.flatData.leadEmail = email;
+          const { leadEmail } = parseProjectToDataObject(project);
+          expect(leadEmail).toEqual(email);
+        });
+
+        test('description is added if available', () => {
+          const expectedDescription = 'this is a description';
+          const project = getGraphQLProject();
+          project.flatData.description = expectedDescription;
+          const { description } = parseProjectToDataObject(project);
+          expect(description).toEqual(expectedDescription);
+        });
+        test('keywords are added', () => {
+          const expectedKeywords = ['Outreach'];
+          const project = getGraphQLProject();
+          project.flatData.keywords = expectedKeywords;
+          const { keywords } = parseProjectToDataObject(project);
+          expect(keywords).toEqual(expectedKeywords);
+        });
+
+        test('keywords are valid', () => {
+          const expectedKeywords = ['invalid-keyword'];
+          const project = getGraphQLProject();
+          project.flatData.keywords = expectedKeywords;
+          expect(() => parseProjectToDataObject(project)).toThrow();
+        });
       });
 
-      test('keywords are valid', () => {
-        const expectedKeywords = ['invalid-keyword'];
-        const project = getGraphQLProject();
-        project.flatData.keywords = expectedKeywords;
-        expect(() => parseProjectToDataObject(project)).toThrow();
+      describe('milestones', () => {
+        test('undefined milestones returns empty array', () => {
+          const Project = getGraphQLProject();
+          Project.flatData.milestones = undefined!;
+          const { milestones } = parseProjectToDataObject(Project);
+          expect(milestones).toEqual([]);
+        });
+        test('the milestones are parsed', () => {
+          const Project = getGraphQLProject();
+          Project.flatData.milestones = [getGraphQLProjectMilestone()];
+          const { milestones } = parseProjectToDataObject(Project);
+          expect(milestones).toEqual([
+            {
+              title: 'A project milestone',
+              status: 'Active',
+            },
+          ]);
+        });
+        test('if present it parses the link', () => {
```

Comment:
same he - i think you could just have theme all in your milestone fixture and then expect them parsed?

Reply:
Link is optional, we should test it separately.